### PR TITLE
add info about workspace services and user resources

### PR DESCRIPTION
**Pull request by TessFerrandez**
# PR for issue #546 

## What is being addressed

Documentation about concepts, authoring and registering workspace services and user resources

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -1,54 +1,96 @@
...
+- [Workspace](#workspace)
+- [Workspace Service](#workspace-service)
+- [User Resource](#user-resource)
+- [Templates](#templates)
 
-A Workspace is a set of resources on a network with inbound traffic, restricted to authorised users, and outbound access restricted to defined network locations. The workspace is a security boundary and there should be zero transfer of data out from the workspace unless explicitly configured. Data transfer is not restricted within a workspace.
+## Application components of the TRE
 
-Workspaces can be configured with a variety of tools to enable tasks such as the development of machine learning models, data engineering, data analysis, and software development.
+A TRE consist of multiple processes orchestrating managing Workspaces and services. These are components that enables Researchers and TRE Admins to provision and manage Workspaces in a self-service manner. These components are of relevance for [Azure Administrators](./user-roles.md#Azure-administrator), [TRE service integrator](./user-roles.md#TRE-service-integrator) and [TRE developers](./user-roles.md#Azure-TRE-developer).
```
Comment (tompaana) [inline docs/concepts.md:25]:
```suggestion
A TRE consist of multiple processes orchestrating managing workspaces and services. These are components that enable researchers and TRE admins to provision and manage workspaces in a self-service manner. The components are of relevance for [Azure administrators](./user-roles.md#Azure-administrator), [TRE service integrator](./user-roles.md#TRE-service-integrator) and [TRE developers](./user-roles.md#Azure-TRE-developer).
```

File: docs/concepts.md [hunk 2]
```diff
@@ -1,54 +1,96 @@
...
+Shared Services are services and resource shared by all workspaces.
 
 - Firewall
 - Package Mirror
 - Git Mirror
 
-### Workspace Services
+## Workspace
+
+A **Workspace** is a set of resources on a network with inbound traffic, restricted to authorised users, and outbound access restricted to defined network locations. The workspace is a security boundary and there should be zero transfer of data out from the workspace unless explicitly configured. Data transfer is not restricted within a workspace.
```
Comment (tompaana) [inline docs/concepts.md:53]:
```suggestion
A **workspace** is a set of resources on a network with inbound traffic, restricted to authorised users, and outbound access restricted to defined network locations. The workspace is a security boundary and there should be zero transfer of data out from the workspace unless explicitly configured. Data transfer is not restricted within a workspace.
```

File: docs/concepts.md [hunk 3]
```diff
@@ -1,54 +1,96 @@
...
+
+The workspace itself contains only the bare essentials to provide this functionality, such as firewalls, storage etc.
+
+Workspaces can be enhanced with one or more building blocks called **Workspace Services** like Azure ML, Guacamole etc. to allow functionality such as development of machine learning models, data engineering, data analysis and software development.
```
Comment (tompaana) [inline docs/concepts.md:57]:
```suggestion
Workspaces can be enhanced with one or more building blocks called **workspace services** like Azure ML, Guacamole etc. to allow functionality such as development of machine learning models, data engineering, data analysis and software development.
```

File: docs/concepts.md [hunk 4]
```diff
@@ -1,54 +1,96 @@
...
+
+Multiple Workspaces can be created within a single Trusted Research Environment to create the required separation for your projects.
```
Comment (tompaana) [inline docs/concepts.md:59]:
```suggestion
Multiple workspaces can be created within a single Trusted Research Environment to create the required separation for your projects.
```

File: docs/concepts.md [hunk 5]
```diff
@@ -1,54 +1,96 @@
...
+
+Each workspace has [workspace users](./user-roles.md): one workspace owner, and one or more workspace researchers that can access the data and workspace services in the workspace. The workspace owner is also considered a workspace researcher.
+
+## Workspace Service
+
+A Workspace Service is a service, created as a building block, with pre-configured set of resources that can be applied to a workspace.
```
Comment (tompaana) [inline docs/concepts.md:65]:
```suggestion
A workspace service is a service, created as a building block, with pre-configured set of resources that can be applied to a workspace.
```

File: docs/concepts.md [hunk 6]
```diff
@@ -1,54 +1,96 @@
...
+Some workspace services, such as Guacamole, allow users to add on user-specific resources (user resources)
 
-### Composition Service
+All workspace services can be deployed to all workspaces.
 
-The Composition Service offers an abstraction over the lower-level Azure resources to allow for TRE users to provision resources in terms of workspaces and workspace services. The Composition Service exposes resources – based on above concepts – as an HTTP API where users and applications can model the desired representation of the TRE, i.e., define which workspaces should contain which workspace services.
-The Composition Service reconciles the desired state with the actual state by invoking Azure resource deployments.
+## User Resource
+
+A User Resource is a resource that is only available to a particular researcher. For example a Guacamole VM.
```
Comment (tompaana) [inline docs/concepts.md:80]:
```suggestion
A user resource is a resource that is only available to a particular researcher. For example a Guacamole VM.
```

File: docs/concepts.md [hunk 7]
```diff
@@ -1,54 +1,96 @@
...
+
+User Resources can be deployed to workspaces with a compatible workspace service. E.g. Guacamole VMs can only be deployed to workspaces where the Guacamole workspace service is deployed.
```
Comment (tompaana) [inline docs/concepts.md:82]:
```suggestion
User resources can be deployed to workspaces with a compatible workspace service. E.g. Guacamole VMs can only be deployed to workspaces where the Guacamole workspace service is deployed.
```

**Review by tompaana — APPROVED** [review]
Looks great! I left some suggestions harmonizing the capitalization. Take'em or leave'em!

**Comment by TessFerrandez** [comment]
> Looks great! I left some suggestions harmonizing the capitalization. Take'em or leave'em!

Thanks much @tompaana - committed all

**Pull request merged**